libs/pageserver_api/src/shard.rs

```diff
@@ -216,7 +230,25 @@ mod tests {
     }
 
     #[test]
-    fn tenant_shard_id_legacy() -> Result<(), hex::FromHexError> {
+    fn tenant_shard_id_binary() -> Result<(), hex::FromHexError> {
+        let example = TenantShardId {
+            tenant_id: TenantId::generate(),
+            shard_count: ShardCount(10),
+            shard_number: ShardNumber(7),
+        };
+
+        let encoded = bincode::serialize(&example).unwrap();
+        let decoded = bincode::deserialize(&encoded).unwrap();
+
+        assert_eq!(example, decoded);
+
+        Ok(())
+    }
+
+    #[test]
+    fn tenant_shard_id_compat_backward() -> Result<(), hex::FromHexError> {
```

Comment:
`tenant_shard_id_backward_compat` reads better I think.

Reply:
I'm ambivalent, but it costs nothing to change it, so changed :-)